scalableminds/webknossos
# Allow Cancelling Jobs

Corresponding PR in worker https://github.com/scalableminds/webknossos-worker/pull/82

New Protocol: 
 - user cancel sets manualState in postgres
 - worker polling gets those with manualState = Cancelled but State != Cancelled (to_cancel)
 - worker terminates those jobs if running
 - worker sends state = Cancelled to wk for those jobs (if it was still running at the time or not)

TODO:
- [x] make latestRunId optional in JobStatus, also in sql
- [x] evolution to add CANCELLED state

### Steps to test:
- set up worker locally, connect to wk via `yarn enable-jobs`
- schedule a job
- in jobs page, click cancel
- should be cancelled (should not be scheduled if it hasn’t already been, should be killed if running)

### Issues:
- fixes #5932 

------
